qabelbox/src/main/java/de/qabel/qabelbox/dagger/modules/MainActivityModule.java

```diff
@@ -39,19 +41,25 @@ public MainActivity provideMainActivity() {
     Identity provideActiveIdentity(IdentityRepository identityRepository,
                                    AppPreference sharedPreferences) {
         String identityKeyId = mainActivity.getIntent().getStringExtra(ACTIVE_IDENTITY);
+        String lastActiveId = sharedPreferences.getLastActiveIdentityKey();
         Identity activeIdentity = null;
         if (identityKeyId != null) {
             try {
                 activeIdentity = identityRepository.find(identityKeyId);
+
+                //Show Toast if active identity is not last active identity.
+                if (activeIdentity != null && lastActiveId != null && lastActiveId.equals(identityKeyId)) {
```

Comment:
Now is the time to extract that code into another class. The dagger modules should really not have so much logic in them, as it is pretty much untestable.

If you just extract this code into another class, inject that instead and maybe IdentityProvider.activeIdentity() on it instead, this code suddenly becomes at least a bit testable and the module gets smaller.